### ISSUE TITLE
Refactoring roadmap

### ISSUE THREAD

**Issue by AdeelH** [azavea/raster-vision#1460]
### for #1395 
- `core`
  - [x] Add a `Labels.from_predictions()` method to easily convert model predictions to `Labels` (#1434)
  - [x] Refactor `ClassConfig` to use pydantic validators (#1436)
  - [x] #1437 
  - [x] #1438 
  - [x] Refactor SegmentationClassTransformer into a RasterTransformer (#1453)
  - [x] Refactor `VectorSource`s (#1461)
    - [x] Decouple class inference from `VectorSource`
    - [x] Refactor `ClassInference` into a `VectorTransformer`
    - [x] Refactor geom buffering into a `VectorTransformer`
- `pytorch_learner`
  - [x] #1413
  - [x] #1434 
  - [x] #1462 
  - [x] Since `Learner` still requires `LearnerConfig`, make defining it as painless as possible
  - [x] Let `Learner.train()` take an `epochs` argument
    - Use case: train for a few epochs --> examine results --> train some more
  - [x] Factor out Visualizer from Learner (Lewis)

### bonus/future
- `core`
  - [ ] (?) Let `VectorSource.get_geojson()` take a `window` argument like `RasterSource.get_chip()`
  - [x] Use geopandas in `VectorSource`, `RasterizedSource`, and `ChipClassificationLabelSource`
  - [x] #1275 
  - [x] Refactor `x_shift`/`y_shift` functionality in `RasterioSource` into a `RasterTransformer`.
  - [x] Simplify MultiRasterSource (remove `raw_channel_order` and `SubRasterSourceConfig` should not be necessary)
  - [x] Allow CRSTransformer to transform windows (not just points)
  - [x]  Remove `ActivateMixin`
  - [ ] (?) Remove EPSG:4326 restriction on GeoJSONs
- `pytorch_learner`
  - ~(?) Remove `overfit_mode`, `test_mode`, and `predict_mode` from `LearnerConfig` for simplicity~
